### PR TITLE
Use overlapping marker spiderfier plugin to spread out markers

### LINKED ISSUE #258: Help spread out markers that overlap (with a plugin)
https://github.com/jawj/OverlappingMarkerSpiderfier-Leaflet does this nicely!

http://jawj.github.io/OverlappingMarkerSpiderfier-Leaflet/demo.html

image

### PR DESCRIPTION
**Pull request by crisner**
Fixes #258  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by crisner** [comment]
Have implemented the feature only on the following layers.
- PFASTracker
- Odorreport
- Mapknitter

The plugin works perfectly on markers that use the default styled icons.

Screenshot of plugin in use on mapknitter layer:
spiderfier-mapknitter

The line when the markers are spiderfied is a bit off on the markers that use custom icons as can be seen in the screenshot of the PFASTracker layer.

Screenshot of plugin in use in the PFASTracker layer:
spiderfier-PFASTracker

There is also a bug where the lines and the popup aren't removed from the view when the layer is turned off.

Screenshot of the bug:
spiderfier-bug

**Comment by crisner** [comment]
@sagarpreet-chadha I haven't found a solution to the bug yet but I shall try looking further. Changing icon anchor seems to help a bit in the case of the problem mentioned in the second screenshot but I did not want to make changes to it without checking with you first. Please review my PR and let me know of things that I need to change, meanwhile I shall try to find a solution to the bug.

**Comment by crisner** [comment]
Fixed bug. Now the spider legs and any active popup on it will get removed when the respective layer is turned off.

File: src/util/spiderfyMarkers.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+omsUtil = function (map, options) {
+    var oms = new OverlappingMarkerSpiderfier(map, options);
+
+    var popup = new L.Popup();
+    oms.addListener('click', function(marker) {
+        popup.setContent(marker._popup._content);
```
Comment (sagarpreet-chadha) [inline src/util/spiderfyMarkers.js]: Are there any leaflet functions to get content from marker?

File: example/layers.js [hunk 2]
```diff
@@ -219,6 +219,11 @@
       "Opensense" : OpenSenseLayer,
       "LSM": OSMLandfillMineQuarryLayer
     };
+
+    var oms = omsUtil(map, {
```
Comment (sagarpreet-chadha) [inline example/layers.js]: Excellent!

**Comment by crisner** [comment]
> Can you verify if this works if there are like 2 layers together on map, for example mapknitter and odorreport (both have oms)

Yes. Please find screenshot attached. The plugin works only on layers that have the `oms.addMarker(marker)` code in it. 

There was one instance just now where I was not able to get the plugin working for some of mapknitter's markers but I assume it was because I started to click on them before they were fully loaded on the page. It started to work as it should when I refreshed the page and haven't been able to reproduce the problem.

Screenshot of the plugin working with two visible layers(mapknitter and odorreport):
twolayers

File: src/util/spiderfyMarkers.js (hunk 1 above)
Comment (crisner) [reply src/util/spiderfyMarkers.js]: I will check the documentation and give you an update.

File: example/layers.js (hunk 2 above)
Comment (crisner) [reply example/layers.js]: Thanks!

File: src/util/spiderfyMarkers.js (hunk 1 above)
Comment (crisner) [reply src/util/spiderfyMarkers.js]: I have refactored the code to make use of the leaflet method to get popup information in my commit 6ecf40f. Thanks for suggesting it!

**Pull request merged**